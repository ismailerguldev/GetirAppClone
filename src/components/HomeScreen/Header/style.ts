import { Dimensions, StyleSheet } from 'react-native'
const { height } = Dimensions.get('window')
export const styles = StyleSheet.create({
    container: {
        backgroundColor: "red",
        justifyContent: "center",
        alignItems: "center"
    },
    logoSection: {
        backgroundColor: "#6a43d9",
        width: "100%",
        height: "100%"
    },
    addressSection: {
        height: height * 0.064,
        backgroundColor: "#F7D102",
        flexDirection: "row"
    },
    addressBox: {
        backgroundColor: "white",
        width: "80%",
        height: "100%",
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "center",
        padding: 15,
        borderTopRightRadius: 30,
        borderBottomRightRadius: 30
    },
    textEv: {
        fontWeight: "bold",
        fontSize: 16
    },
    tvsBox: {
        justifyContent: "center",
        alignItems: "center",
        width: "20%"
    },
    textTvs: {
        fontSize: 12,
        color: "#6a43d9",
        fontWeight:"bold"
    },
    textDk: {
        fontSize: 24,
        color: "#6a43d9",
        fontWeight:"bold"
    }
})