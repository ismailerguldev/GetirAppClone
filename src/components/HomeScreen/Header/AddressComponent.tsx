import { Text, View, Image } from 'react-native'
import Entypo from '@expo/vector-icons/Entypo';
import React from 'react'
import { styles } from './style';
const AddressComponent = () => {
    return (
        <View style={styles.addressSection}>
            <View style={styles.addressBox}>
                <Image source={{ uri: "https://cdn.getir.com/misc/emoji/house.png" }} width={30} height={30} />
                <Text style={styles.textEv}>Ev</Text>
                <Text>Dedepaşa Blv. Yenişehir Mahallesi...</Text>
                <Entypo name="chevron-right" size={24} color="#6a43d9" />
            </View>
            <View style={styles.tvsBox}>
                <Text style={styles.textTvs}>TVS</Text>
                <Text style={styles.textDk}>13dk</Text>
            </View>
        </View>
    )
}

export default AddressComponent