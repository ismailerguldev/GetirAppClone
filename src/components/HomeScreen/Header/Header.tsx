import { StyleSheet, Text, View } from 'react-native'
import React from 'react'
import AddressComponent from './AddressComponent'

const Header = () => {
  return (
    <View testID='HeaderComponent'>
      <AddressComponent />
    </View>
  )
}

export default Header

const styles = StyleSheet.create({})