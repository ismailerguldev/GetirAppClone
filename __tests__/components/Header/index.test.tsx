import React from "react";
import { render, waitFor } from "@testing-library/react-native"
import Header from "../../../src/components/HomeScreen/Header/Header";
describe("components/HomeScreen/Header", () => {
    test("renders Correctly", async () => {
        const { toJSON, debug } = render(<Header />)
        expect(toJSON()).toMatchSnapshot()
    })
})