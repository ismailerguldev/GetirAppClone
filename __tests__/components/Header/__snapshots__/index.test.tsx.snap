// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/HomeScreen/Header renders Correctly 1`] = `
<View
  testID="HeaderComponent"
>
  <View
    style={
      {
        "backgroundColor": "#F7D102",
        "flexDirection": "row",
        "height": 85.376,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomRightRadius": 30,
          "borderTopRightRadius": 30,
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "space-between",
          "padding": 15,
          "width": "80%",
        }
      }
    >
      <Image
        height={30}
        source={
          {
            "uri": "https://cdn.getir.com/misc/emoji/house.png",
          }
        }
        width={30}
      />
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Ev
      </Text>
      <Text>
        Dedepaşa Blv. Yenişehir Mahallesi...
      </Text>
      <Text />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "20%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#6a43d9",
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
      >
        TVS
      </Text>
      <Text
        style={
          {
            "color": "#6a43d9",
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        13dk
      </Text>
    </View>
  </View>
</View>
`;
